src/GridFS/StreamWrapper.php

```diff
@@ -280,8 +292,24 @@ private function getStatTemplate(): array
      */
     private function initProtocol(string $path): void
     {
-        $parts = explode('://', $path, 2);
-        $this->protocol = $parts[0] ?: 'gridfs';
+        $parts = parse_url($path);
```

Comment:
I don't think this is a good use case for `parse_url()`. You only access `scheme` and `path` after this point, and `parse_url()` also captures the database name in a `host` component, which seems odd.

If we parse a typical readable path of `gridfs://{db}/{coll}.files/{id}`, the `path` component will be `/{coll}.files/{id}`. You'll later explode that to yield an array with three elements: `['', '{coll}.files', '{id}']`. The later logic to explode the `path` and extract the ID works fine, but it seems unnecessary.

The `path` component is only relevant if we enter the branch where we have no context options.

I'd suggest reverting this to explode once on `://` to extract the protocol. From there, you can then explode again if needed to extract the ID, and just account for the database component appearing in the path. I actually don't think the second `explode()` would change at all, since there'd be no leading slash in `$parts[1]` after the initial split.

Reply:
No longer using `parse_url`. The filename is extracted after the default "base path" of the bucket.